core/src/luigi-config/utils/params-version.js

```diff
@@ -0,0 +1,23 @@
+import { major, satisfies, coerce } from 'semver';
+
+const PARAMS_VERSION = '1.0'; // make sure to sync it in core-ui
+
+export function areParamsCompatible(paramsVersion) {
+  const majorVersion = major(coerce(PARAMS_VERSION));
+  return satisfies(coerce(paramsVersion), `^${coerce(majorVersion)}`);
+}
+
+export function showIncompatibleParamsWarning(paramsVersion) {
+  const message = `Configuration incompatible: version: ${paramsVersion ||
+    "'unknown'"}, supported version: ${PARAMS_VERSION}. Errors may occur.`;
```

Comment:
Maybe something like this since we support all minor versions?
```suggestion
    "'unknown'"}, supported versions: ^${PARAMS_VERSION}. Errors may occur.`;
```